packages/twenty-front/src/modules/settings/data-model/objects/forms/components/SettingsDataModelObjectIdentifiersForm.tsx

```diff
@@ -65,41 +71,37 @@ export const SettingsDataModelObjectIdentifiersForm = ({
       {[
         {
           label: 'Record label',
-          fieldName: 'labelIdentifierFieldMetadataId' as const,
+          fieldName: LABEL_IDENTIFIER_FIELD_METADATA_ID,
           options: labelIdentifierFieldOptions,
+          defaultValue:
+            objectMetadataItem[LABEL_IDENTIFIER_FIELD_METADATA_ID] ??
+            defaultLabelIdentifierFieldMetadataId, // defaultLabelIdentifierFieldMetadataId could be undefined as it's injected as any
```

Comment:
// defaultLabelIdentifierFieldMetadataId could be undefined as it's injected as any